service/history/api/respondworkflowtaskcompleted/api.go

```diff
@@ -506,10 +506,9 @@ func (handler *WorkflowTaskCompletedHandler) Invoke(
 		}
 	}
 
-	bypassTaskGeneration := request.GetReturnNewWorkflowTask() && wtFailedCause == nil
-	// TODO (alex-update): All current SDKs always set ReturnNewWorkflowTask to true
-	//  which means that server always bypass task generation if WFT didn't fail.
-	//  ReturnNewWorkflowTask flag needs to be removed.
+	// By pass normal task generation flow and return the next workflow task as part of the response.
+	// It is efficient to bypass unless the task had failed.
+	bypassTaskGeneration := handler.shouldBypassTaskGeneration(req, wtFailedCause)
```

Comment:
not sure I agree with the variable and method name as L525 - L543 are overriding this bool flag as well. 

Not blocking but please help see if they can be merged somehow.